File: src/gestorAplicacion/gente/Empleado.java
package gestorAplicacion.gente;
import gestorAplicacion.restaurante.*;
import java.util.ArrayList; 
import java.util.Collections;
import java.io.Serializable;

public class Empleado extends Persona implements Serializable {
private int numero_ventas;
private int numero_asignado; //el numero asignado al empleado
public static ArrayList<Empleado> lista_empleados = new ArrayList<Empleado>();
public static int numero_empleados;

//constructor

public Empleado(int cedula,String nombre,int telefono) {
	super(cedula,nombre,telefono);
	Empleado.lista_empleados.add(this);
	numero_empleados++;
	numero_asignado=numero_empleados;
}

//metodos get y set

public int getNumero_asignado(){
	return numero_asignado;
}

public void setNumero_asignado(int numero_asignado){
	this.numero_asignado=numero_asignado;
}

public static int getNumero_empleados() { 
	return numero_empleados;
}

public int getNumero_ventas() {
	return numero_ventas;
}


// otros metodos

public void nuevaVenta() {
	numero_ventas++;
}

public static Empleado empleado_mas_eficiente() { //este metodo compara la cantidad de ventas de los empleados y devuelve el que MAS ventas tiene
	ArrayList<Integer> cantventasporempleado = new ArrayList<Integer>(); //arraylist con la cantidad de ventas por empleado
	for (int l=0;l<Empleado.lista_empleados.size();l++) {
		cantventasporempleado.add(lista_empleados.get(l).getNumero_ventas());
	}
	Integer maximo = Collections.max(cantventasporempleado); //maximo numero de ventas por empleado
	for (int j=0;j<cantventasporempleado.size();j++) {
		if(cantventasporempleado.get(j)==maximo) {
			return lista_empleados.get(j);
		}
	}
	return null;	
}


public static Empleado empleado_menos_eficiente() { //este metodo compara la cantidad de ventas de los empleados y devuelve el que MENOS ventas tiene
	ArrayList<Integer> cantventasporempleado = new ArrayList<Integer>(); //arraylist con la cantidad de ventas por empleado
	for (int l=0;l<Empleado.lista_empleados.size();l++) {
		cantventasporempleado.add(lista_empleados.get(l).getNumero_ventas());
	}
	Integer minimo = Collections.min(cantventasporempleado); //minimo numero de ventas por empleado
	for (int j=0;j<cantventasporempleado.size();j++) {
		if(cantventasporempleado.get(j)==minimo) {
			return lista_empleados.get(j);
		}
	}
	return null;	
}

public String informacion() {
	return "nombre del Empleado: "+this.getNombre()+" \ncedula: "+this.getCedula()+" \ntelefono: "+this.getTelefono()+" \nnumero asignado: "+numero_asignado+" \nnumero de ventas: "+numero_ventas;
}

public String info_basi() {
	return super.informacion();
}

public String perfil() {
	if(this.getNumero_ventas()>25) {
		return "tiene mas de 50 ventas, felicidades";
	}
	else {
		return "tiene menos de 50 ventas, debe esforzarce mas";
	}
}


}

File: Python/gestorAplicacion/gente/empleado.py
from csv import list_dialects
from persona import Persona
class Empleado(Persona):
    numero_empleados=0
    lista_empleado=[]
    def __init__(self,cedula,nombre,telefono):
        super.__init__(self,cedula,nombre,telefono)
        self.numero_ventas=0
        numero_empleados+=1
        self._numero_asignado=numero_empleados
        self._numero_ventas=0
        lista_empleado.append(self)
    
    #metodos get
    def getNumero(self):
        return self._numero_asignado
    
    @classmethod
    def getNumero_empleados(self):
        return numero_empleados
    
    def getNumero_ventas(self):
        return self._numero_ventas
    
    #metodos set
    def setNumero_asignado(self,numero_asignado):
        self._numero_asignado=numero_asignado
    
    #otros metodos
    def nuevaVenta(self):
        self._numero_ventas+=1
    
    @classmethod
    def empleado_mas_eficiente(cls):
        ventasporempleado=[]
        for i in lista_empleado:
            ventasporempleado.append(i.getNumero_ventas())
        maximo_ventas=max(ventasporempleado)
        for e in ventasporempleado:
            if ventasporempleado(e)==maximo_ventas:
                return lista_empleado[e]
    
    @classmethod
    def empleado_menos_eficiente(cls):
        ventasporempleado=[]
        for i in lista_empleado:
            ventasporempleado.append(i.getNumero_ventas())
        minimo_ventas=min(ventasporempleado)
        for e in ventasporempleado:
            if ventasporempleado(e)==minimo_ventas:
                return lista_empleado[e]
    
    def informacion(self):
        return f"nombre del Empleado: {self._nombre} \ncedula: {self._cedula} \ntelefono: {self._telefono}\nnumero asignado: {self._numero_asignado} \nnumero de ventas: {self._numero_ventas}"
    
    def info_basi(self):
	    super().informacion()